controllers/awsmachinetemplate_controller.go

```diff
@@ -160,6 +160,34 @@ func (r *AWSMachineTemplateReconciler) reconcileDelete(ctx context.Context, iamS
 			}
 		}
 	}
+
```

Comment:
What does this change do? Please add a PR description why this fixes something, and in which problem case.

This can introduce a new problem: if the `AWSCluster` finalizer (now removed _last_) failed to be removed, then the next reconciliation may fail to read the `ConfigMap` because its finalizer was already successfully removed.